### PR TITLE
DLC TLVs

### PR DESCRIPTION
**Pull request by nkohen**
Introduces TLVs for `DLCOffer`, `DLCAccept`, and `DLCSign` as specified in https://github.com/discreetlogcontracts/dlcspecs/blob/cf7e49e6a852c99e076777fbb96142e626cac641/Protocol.md

### CODE REVIEW

**Comment by benthecarman** [comment]
doesn't compile

File: crypto/src/main/scala/org/bitcoins/crypto/ECDigitalSignature.scala [hunk 1]
```diff
@@ -121,6 +121,16 @@ object ECDigitalSignature extends Factory[ECDigitalSignature] {
     }
   }
 
+  def fromFrontOfBytes(bytes: ByteVector): ECDigitalSignature = {
+    val sigWithExtra = fromBytes(bytes)
+    val sig = fromRS(sigWithExtra.r, sigWithExtra.s)
+
+    require(bytes.startsWith(sig.bytes),
+            s"Received weirdly encoded signature at beginning of $bytes")
+
```
Comment (benthecarman) [inline crypto/src/main/scala/org/bitcoins/crypto/ECDigitalSignature.scala:130]: Should you require that `sig == sigWithExtra`?

**Review by benthecarman — COMMENTED** [review]
lgtm otherwise

File: crypto/src/main/scala/org/bitcoins/crypto/ECDigitalSignature.scala (hunk 1 above)
Comment (nkohen) [reply crypto/src/main/scala/org/bitcoins/crypto/ECDigitalSignature.scala:130]: no, `sigWithExtra` has all the bytes

**Pull request merged**